### PR TITLE
async delete files [#4065]

### LINKED ISSUE #4065: Deleting 3 Files Takes Up to 12 Seconds
Steps to reproduce:

1. Create a new resource
2. Upload a tiny text file, maybe 113 bytes
3. Create external content with "add content from the web"
4. Create a folder named "Test"
5. Enter the folder
6. Create external content with "add content from the web"
7. Return to root folder with the up arrow browser
8. Delete the folder and two files
9. Observe wait time

### PR DESCRIPTION
**Pull request by devincowan**
Resolves #4065 

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. Upload a bunch of small files to a resource
2. Delete a handful of them
3. Note that you can still interact with the page and the deletion is done asynchronously

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4615](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4615)

**Comment by devincowan** [comment]
@Maurier I didn't see a good way to speed up file deletion. But at least we can provide UI feedback and do it async without page reload. LMK if I missed an obvious way to speed things up...thanks

**Comment by devincowan** [comment]
@Maurier I didn't see a good way to speed up file deletion. But at least we can provide UI feedback and do it async without page reload. LMK if I missed an obvious way to speed things up...thanks

**Review by Maurier — CHANGES_REQUESTED** [review]
Looks good. Left a minor suggestion to improve UX.

**Comment by Maurier** [comment]
> @Maurier I didn't see a good way to speed up file deletion. But at least we can provide UI feedback and do it async without page reload. LMK if I missed an obvious way to speed things up...thanks

The frontend already sends all the file ids to delete in one request to the backend. So, any improvement that could be done has to look into the backend implementation of files deletion. But I think the work done in this PR effectively covers the scope outlined here.

File: theme/static/css/hydroshare_core.css [hunk 1]
```diff
@@ -4296,6 +4296,10 @@ ul#fb-files-container .hs-upload-indicator {
 	pointer-events:none;
 }
 
+#fb-files-container li.deleting {
```
[the remark this answers is not in the document]
Comment (devincowan) [reply theme/static/css/hydroshare_core.css:4299]: @Maurier thanks! I actually had `pointer-events: none;` in there but reverted it when I added the `deleteList.css("cursor", "wait")` but you're right, the **wait** doesn't stop events from propagating in our FB, so good call, thank you!

**Review by Maurier — APPROVED** [review]
Looks good

**Comment by devincowan** [comment]
sorry @Maurier I found one more small improvement after your approval :( [634e158](https://github.com/hydroshare/hydroshare/pull/4615/commits/634e1589db3af37869c77fbaa3c424a8cfa29801)

**Review by Maurier — APPROVED** [review]

**Pull request merged**